### ISSUE TITLE
[master] Items that character can no longer use.

### ISSUE THREAD

**Issue by Eredar** [TrinityCore/TrinityCore#18548]
[//]: # (***************************************************************)
[//]: # (** DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED **)
[//]: # (***************************************************************)

**Description:** Items that character can no longer use should be removed to inventory instantly. 

**Current behaviour:** Currently you can use an item, even if it has wrong criterias of usage.

For example: you are Arms Warrior and using Strom'kar, arms warrior two-handed artifact sword that has Arms Spec requirement. After you will change your spec, you still will be able to use it in Fury until first loading screen. After it, item will be removed and sended to you via mail. Btw sword has lost all assigned traits and artifact power after the mail, but it is different issue (or maybe not, I didn't use to get my artifact by the mail on live servers).

Also artifact should be autoequipped respectively when you change your spec, if it's in the inventory (maybe it is different issue too).

Other (hack) example: You are level 110 character, wearing lvl 110 items. If you use .levelup -10 or any other negative value to become lvl 100 for example, you will still be able to use equipped items (that require level 110) and have ridiculous amounts of crit/haste/mastery/versality percents. I saw 300% haste, 150% of crit, etc. And again items will be removed only after first loading screen.

**Expected behaviour:** As soon as equipping criterias are failed, items should be removed to inventory. 

**Steps to reproduce the problem:**

1. Explaned above, but easiest way to demostrate: Create a character.
2. Use .levelup 109, use .add 140808, equip it.
3. Use .levelup -100, you will be able to use the trinket until first loading screen.
4. More complicated and intended way explained in the first example above.

**Branch(es):** master

**TC rev. hash/commit:**  538c8c3843fdf8705e348960d19b279653c03b95

**TDB version:**  TDB 703.00 with custom patches.

**Operating system:**  Win 10 x64 Pro

[//]: # (This template is for problem reports. For other types of report, edit it accordingly.)
[//]: # (If this is a crash report, include the crashlog with https://gist.github.com/)
[//]: # (For fixes containing C++, create a Pull Request.)

**Comment by Shauren** [comment]
No, they should not. The only feature you are seeing here is that on retail artifacts are autoequipped when you change specs.
For example: if you have a druid in balance spec with balance artifact equipped and change spec to feral but you didn't complete feral artifact questline (so you don't have it) then balance artifact will not be unequipped.

**Comment by Eredar** [comment]
Ok, maybe I'm wrong with "unequipped" part, but on live core makes them "red", not usable in other spec, as if they have no durability. And in actual state of tc you can use them untill first loading screen.
